### ISSUE TITLE
crossgen'ed test app hangs with corerun 

### ISSUE THREAD

**Issue by desunit** [dotnet/coreclr#1634]
I've been playing with `crossgen` to generate `ni.exe` file for a simple test application but noticed that crossgened apps just hangs with `corerun`. The strange thing is that test application works again once I delete `test.ni.exe`. The `test.cs` is:

```
using System;

public class Program
{
    public static void Main (string[] args)
    {
        var a = 10;
        //System.Console.WriteLine("Hello World!");
    }
}
```

I use the latest CoreCLR code compiled with XCode7.

**Comment by jkotas** [comment]
I am not able to reproduce the hang. Could you please break in with debugger and get the stack of the hang? It should give us some hints of what may be going on.

**Comment by desunit** [comment]
Thanks for reply! I've just updated to the latest Mono (4.0.4.0) and it looks everything is working fine. Though I've noticed that I can reproduce it with Exception:
```
public class Program
{
    public static int Main (string[] args)
    {
        throw new Exception("Test!");
    } 
}
```

Without crossgen I see:
```

Unhandled Exception: System.Exception: Test!
   at Program.Main(String[] args)
```

then I've added:
```
./crossgen test.exe
echo Running test.exe
./corerun -c $CORECLR test.exe
```

and received the following output:

```
Microsoft (R) CoreCLR Native Image Generator - Version 4.5.22220.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Native image /Volumes/EXTDISK/Research/llilc/coreclr/bin/Product/OSX.x64.Debug/test.ni.exe generated successfully.
Running test.exe
```

**Comment by jkotas** [comment]
@janvorli Could you please take a look? (Looks like an issue with the unwinder.)

**Comment by janvorli** [comment]
@jkotas sure

**Comment by janvorli** [comment]
Fixed by #1661
